### ISSUE TITLE
Fully Functional Tauri Client

### ISSUE THREAD

**Issue by nothingismagick** [padloc/padloc#281]
We would like to help make sure that Tauri builds properly, especially because we want to start using this sweet piece of awesomeness ourselves. It would include a proper GH Actions integration - [see this repo](https://github.com/tauri-apps/tauri-action). 

If you are so inclined, please feel invited get in touch on our discord, because we are considering making this project a new "special focus" project where the core team makes sure that your problems are not only resolved, but that your needs are met - even if that means API revisions or entirely new interfaces. Obviously we will only do this if you are interested.

At any rate, kudos and let's see how we can collaborate.

**Comment by nothingismagick** [comment]
Check this out: 
https://tauri.studio/docs/usage/recipes/plugin/#writing-a-plugin (specifically the `init_script` part)

**Comment by MaKleSoft** [comment]
@nothingismagick Thank you so much for reaching out! I was actually planning to get in touch with you to suggest some kind of collaboration, too 😄 Tauri looks like perfect fit for Padloc and I believe Padloc could be great proof of concept for Tauri in turn, especially considering your security-centric approach. Replacing our Electron and Cordova wrappers with Tauri could be a huge win for both security and performance and would allow us to streamline our build pipeline significantly. I'll be reaching out on Discord soon!

**Comment by MaKleSoft** [comment]
I haven't looked at the Tauri integration in a while, but last time I checked there where still the following blockers:

- [ ] **Clipboard support**: The [Tauri roadmap](https://tauri.studio/en/#roadmap) says that clipboard support is still to come, but I'm hoping it's just not up to date. Hopefully just adding an application menu with the appropriate items will do the trick. Related issues: https://github.com/tauri-apps/tauri/issues/1055 https://github.com/tauri-apps/tauri/issues/2397. See also: https://tauri.studio/en/docs/usage/guides/visual/menu. We're also using [`document.execCommand("cut")`](https://github.com/padloc/padloc/blob/master/packages/app/src/lib/platform.ts#L37) and [`document.execCommand("paste")`](https://github.com/padloc/padloc/blob/master/packages/app/src/lib/platform.ts#L48) to access the clipboard programatically, so we'll have to check if that works in Tauri (it does in electron).
- [ ] **Storage**: Padloc uses [localForage](https://github.com/localForage/localForage) for local storage. Last time I checked there were problems with using "vanilla" localForage, but luckily there seems to be a plugin for that now: https://github.com/tauri-apps/tauri-forage
- [ ] **Code Signing**: This seems to be a [solved](https://tauri.studio/en/docs/usage/guides/bundler/sign-osx) for macos. Windows will be more tricky to achieve via CI (but that's no different from Electron).
- [ ] **Auto Updates**: This was a pain to get working properly with Electron. Let's hope it is easier with Tauri: https://tauri.studio/en/docs/usage/guides/updater

**Comment by nothingismagick** [comment]
- copy-paste works. not sure about `document.execCommand` though. will need to be tested cross-platform
- i am not sure tauri-forage will be a drop-in-replacement for you here. we would probably recommend using the `tauri-plugin-stronghold` instead.
- code-signing works
- auto updates should also work

**Comment by MaKleSoft** [comment]
@nothingismagick Thanks for the input! Looks like we've got a working beta build now! There are still a couple of things to work out, but we'll track these in separate tickets:

- Auto updates: https://github.com/padloc/padloc/issues/365
- Clipboard: Turns out the only thing that wasn't working were keyboard bindings on MacOS, and that should be fixed by adding an application menu, which is tracked here: https://github.com/padloc/padloc/issues/364
- localForage is actually working fine for now, but we'll still consider moving to Stronghold in the near future: https://github.com/padloc/padloc/issues/366

Closing this!
